compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUConfigureTensorLayouts.cpp

```diff
@@ -0,0 +1,133 @@
+                                   linalg::LinalgOp contract) {
+  // TODO: Add SIMT fallback.
+  if (!schedule) {
+    return contract->emitError("missing mma schedule for contraction");
+  }
+
+  // This function should have only be called on a contraction op.
+  assert(linalg::isaContractionOpInterface(contract) &&
+         "cannot set contraction anchor on non contraction op");
+
+  auto layouts = schedule.getContractionLayout(contract);
+  if (failed(layouts)) {
+    return contract->emitError("cannot get concrete layout for contraction");
+  }
+
+  auto [aLayout, bLayout, cLayout] = *layouts;
+  Location loc = contract.getLoc();
+
+  Value lhs = contract->getOperand(0);
+  Value rhs = contract->getOperand(1);
+  Value acc = contract->getOperand(2);
+
+  // Set layouts for lhs, rhs and acc.
+  rewriter.setInsertionPoint(contract);
+  auto layoutedLhs =
+      rewriter.create<IREE::VectorExt::ToLayoutOp>(loc, lhs, aLayout);
+  auto layoutedRhs =
+      rewriter.create<IREE::VectorExt::ToLayoutOp>(loc, rhs, bLayout);
+  auto layoutedAcc =
+      rewriter.create<IREE::VectorExt::ToLayoutOp>(loc, acc, cLayout);
+
+  // Promote matmul lhs and rhs.
+  // TODO: We should read this from the lowering_config on the operation.
+  // TODO: This is a hack until layout analysis is improved. The layout analysis
+  // should decide where to put these shared memory conversions.
+  layoutedLhs.setSharedMemoryConversion(true);
+  layoutedRhs.setSharedMemoryConversion(true);
+
+  contract->setOperand(0, layoutedLhs.getResult());
+  contract->setOperand(1, layoutedRhs.getResult());
+  contract->setOperand(2, layoutedAcc.getResult());
+
+  // Set layout for result.
+  rewriter.setInsertionPointAfter(contract);
+  auto toLayout = rewriter.create<IREE::VectorExt::ToLayoutOp>(
+      loc, contract->getResult(0), cLayout);
+  rewriter.replaceAllUsesExcept(contract->getResult(0), toLayout.getResult(),
+                                toLayout);
+
+  return success();
+}
+
+struct LLVMGPUConfigureTensorLayoutsPass final
+    : impl::LLVMGPUConfigureTensorLayoutsPassBase<
+          LLVMGPUConfigureTensorLayoutsPass> {
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<IREE::VectorExt::IREEVectorExtDialect>();
+    registry.insert<vector::VectorDialect>();
+  }
+
+  void runOnOperation() override {
+    auto func = getOperation();
+
+    std::array<int64_t, 3> workgroupSize;
+    std::optional<SmallVector<int64_t>> maybeWorkgroupSize =
+        getWorkgroupSize(func);
+    if (!maybeWorkgroupSize) {
+      func->emitOpError()
+          << "unable to query workgroup_size information from entry point";
+      return signalPassFailure();
+    }
+    for (auto [index, value] : llvm::enumerate(maybeWorkgroupSize.value())) {
+      workgroupSize[index] = value;
+    }
+    for (auto index : llvm::seq<size_t>(maybeWorkgroupSize->size(), 3)) {
+      workgroupSize[index] = 1;
```

Comment:
Are we are copying to an array, and then replacing the values with 1s? 
Also where is this workgroupSize being used?